core/src/main/java/org/apache/druid/common/exception/AllowedRegexErrorResponseTransformStrategy.java

```diff
@@ -51,7 +51,8 @@ public AllowedRegexErrorResponseTransformStrategy(
   public Function<String, String> getErrorMessageTransformFunction()
   {
     return (String errorMessage) -> {
-      if (allowedRegexPattern.stream().anyMatch(pattern -> pattern.matcher(errorMessage).matches())) {
+      if (errorMessage == null || allowedRegexPattern.stream()
```

Comment:
Feels a little weird because the user has explicitly specified the exact patterns that they want to allow. I wonder if we could have a pattern that matches `null`, but I suppose that wouldn't be possible.

Maybe we should call it out in the class javadoc that we will always allow `null`.

Reply:
Great point. We'd have to agree on the meaning of a null message. It might make sense to map null messages to an empty message, then apply the regex rules as normal.

Even further back, is a null message helpful? If I get an error that says `{errorMessage: null}`, I'm not going to be in a good position to figure out what's wrong. Should we fix the issue upstream, where something is providing an unhelpful (null) message?

Reply:
Here's what I was thinking: in the current implementation, if the regex doesn't match then we return null. If it does match, we return the original message. If the original message is null itself, these outcomes are indistinguishable and so there is no need to apply the regex. 

As to whether a null message is helpful or not: no, not really, but the interface allows it and there's some cases where it is in fact null. I do not have the time right now to go find and change them all. But I do have time to fix this class to conform to the current behavior of the interface. Therefore I suggest we do this fix, and scrub through and improve the error messages as time allows.

Reply:
Sounds good.